internal/initializer/initializer.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package initializer
+
+import (
+	"context"
+
+	"github.com/crossplane/crossplane/apis"
+
+	"github.com/pkg/errors"
+	extv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/crossplane/crossplane-runtime/pkg/resource"
+)
+
+// NewInitializer returns a new *Initializer.
+func NewInitializer(steps ...Step) *Initializer {
+	return &Initializer{Steps: steps}
+}
+
+// TODO(muvaf): We will have some options to inject CA Bundles to those CRDs
+// before applying them.
+
+// Step is a blocking step of the initialization process.
+type Step interface {
+	Run(ctx context.Context, kube resource.ClientApplicator) error
+}
+
+// Initializer makes sure the CRDs Crossplane reconciles are ready to go before
+// starting main Crossplane routines.
+type Initializer struct {
+	Steps []Step
+}
```

Comment:
Nit - could this type be a slice? e.g.:

```go
type Initializer []Step
```

Reply:
I actually made some changes to this. It used to figure out the client config on its own, now `Initializer` structs needs a client passed in. So, now it has another field.